internal/imagedefinition/README.rst

```diff
@@ -258,6 +258,14 @@ The following specification defines what is supported in the YAML:
                name: <string>
                # The UID to assing to this new user
                id: <string> (optional)
+               # Password. This can be a plain text or a hashed value.
+               password: <string> (optional)
+               # Type of password submitted above. Defaults to "hash" 
+               password-type: text | hash (optional)
+               # Should the password immediately expire and force the user to 
+               # renew it at first login
+               # Defaults to true
+               expire: <boolean>
```

Comment:
It's good that we're giving such full flexibility, but I'm just wondering if maybe we should just skip giving users the ability to build hard-coded user/password images which do not have expiring-on-first-login users. Did we get a request for this field? Maybe we should just default here with `true` and not define it in the yaml.

Reply:
I fully agree that from a security point-of-view that would be better to force password reset.  But since this feature was requested to replace a call to cloud-init I thought it best to try and reproduce the cloud-init feature. In the LP big Ethan requested that we should be able to force the password to be reset at first login. So indeed we can avoid this field and always force the reset but I suspect we will have to rework this at some point for users requesting building images with non-expiring password.

Reply:
I'm still a bit conflicted here. I'm just a bit worried that we'll end up with some users building images with non-expiring passwords hard-coded in their yaml files that are made publicly available and then basically shipping this out to some customers. Then again... is this possible with cloud-init actually?

This is why I try to have the mentality "let's play it safe and only discuss introduction if the feature is requested".